src/classes/CON_DoNotContact_TDTM.cls

```diff
@@ -32,95 +32,177 @@
+            this.syncDoNotContactFieldsOnContacts(deceasedContactsList);
+        }
+
+        if (doNotContactContactsList.isEmpty() == false){
+            this.syncDoNotContactFieldsOnContacts(doNotContactContactsList);
+        }
     }
 
-    /*******************************************************************************************************
-    * @description Copies the values from the naming exclusions checkboxes to the multiselect picklist.
-    * @param con The Contact
-    * @return void
-    ********************************************************************************************************/
-    public static void copyNamingExclusionsFromCheckboxes(Contact con) {
-        string mspValue = '';
+    /**
+     * @description Synchronizes the communication preference fields
+     * (Do not Call, Email Opt Out, Fax Opt Out) on a contact record
+     * to match the status specified by the "Do Not Contact" field.
+     * @param contactsList List of contacts to synchronize
+     */ 
+    @TestVisible
+    private void syncDoNotContactFieldsOnContacts(List<Contact> contactsList){
+        for (Contact con : contactsList){
+            con.HasOptedOutOfEmail = con.Do_Not_Contact__c;
+            con.DoNotCall = con.Do_Not_Contact__c;
+            con.HasOptedOutOfFax  = con.Do_Not_Contact__c;
+        }
+    }
 
-        if (con.Exclude_from_Household_Name__c)
-            mspValue += 'Household__c.Name;';
-        if (con.Exclude_from_Household_Formal_Greeting__c)
-            mspValue += 'Household__c.Formal_Greeting__c;';
-        if (con.Exclude_from_Household_Informal_Greeting__c)
-            mspValue += 'Household__c.Informal_Greeting__c;';
+    /**
+     * @description Synchronizes the Household Naming Exclusion fields 
+     * (checkboxes and picklist) on a contact record to match the status 
+     * specified by the "Deceased" field.
+     * @param contactsList List of contacts to synchronize
+     */ 
+    @TestVisible
+    private void syncNamingExclusionsOnContacts(List<Contact> contactsList){
+        for (Contact con : contactsList){
+            con.Exclude_from_Household_Name__c = con.Deceased__c;
+            con.Exclude_from_Household_Formal_Greeting__c = con.Deceased__c;
+            con.Exclude_from_Household_Informal_Greeting__c = con.Deceased__c;
+        }
 
-        if (mspValue.endsWith(';'))
-            mspValue = mspValue.substring(0,mspValue.length()-1);
+        this.syncHHNamingExclusionsPicklistFromCheckboxes(contactsList);
+    }
 
-        con.Naming_Exclusions__c = mspValue;
+    /**
+     * @description Synchronizes the naming exclusions specified by the selected 
+     * checkboxes with the selected values listed in the Naming Exclusion 
+     * multiselect picklist.
+     * @param contactsList List of contacts to synchronize
+     */ 
+    @TestVisible
+    private void syncHHNamingExclusionsPicklistFromCheckboxes(List<Contact> contactsList) {
```

Comment:
Naming is hard. I am wondering if we can avoid using generic phrases like 'FromCheckboxes'. People might ask what checkboxes and get confused.